code/messaging/src/main/java/com/adobe/marketing/mobile/MessagingPushBuilder.java

```diff
@@ -0,0 +1,196 @@
+                return channelIdFromPayload;
+            } else {
+                Log.debug(MessagingPushConstants.LOG_TAG, SELF_TAG, "Channel does not exist for channel ID obtained from payload ( " + channelIdFromPayload + "). Using the Messaging Extension's default channel.");
+            }
+
+            // Use the default channel ID if the channel ID from the payload is null or if a channel does not exist for the channel ID from the payload.
+            final String channelId = DEFAULT_CHANNEL_ID;
+            if (notificationManager.getNotificationChannel(DEFAULT_CHANNEL_ID) != null) {
+                Log.debug(MessagingPushConstants.LOG_TAG, SELF_TAG, "Channel already exists for the default channel ID: " + channelId);
+                return DEFAULT_CHANNEL_ID;
+            } else {
+                Log.debug(MessagingPushConstants.LOG_TAG, SELF_TAG, "Creating a new channel for the default channel ID: " + channelId + ".");
+                final NotificationChannel channel = new NotificationChannel(channelId, DEFAULT_CHANNEL_NAME, NotificationManager.IMPORTANCE_DEFAULT);
+                notificationManager.createNotificationChannel(channel);
+            }
+            return channelId;
+        }
+    }
+
+
+    private void addActionButtons(final NotificationCompat.Builder builder) {
+        // TODO: Add this in the future Pull Request
+    }
+
+    /**
+     * Sets the small icon for the notification.
+     * If a small icon is received from the payload, the same is used.
+     * If a small icon is not received from the payload, we use the icon set using MobileCore.setSmallIcon().
+     * If a small icon is not set using MobileCore.setSmallIcon(), we use the default small icon of the application.
+     *
+     * @param builder the notification builder
+     */
+    private void setSmallIcon(final NotificationCompat.Builder builder) {
+        final int iconFromPayload = utils.getSmallIconWithResourceName(payload.getIcon());
+        final int iconFromMobileCore =  -1 ;   //MobileCore.getSmallIconResourceId();
+
+        if (isValidIcon(iconFromPayload)) {
+            builder.setSmallIcon(iconFromPayload);
+        } else if (isValidIcon(iconFromMobileCore)) {
+            builder.setSmallIcon(iconFromMobileCore);
+        } else {
+            final int iconFromApp = utils.getDefaultAppIcon();
+            if (isValidIcon(iconFromApp)) {
+                builder.setSmallIcon(iconFromApp);
+            } else {
+                Log.warning(MessagingPushConstants.LOG_TAG, SELF_TAG, "No valid small icon found. Notification will not be displayed.");
+            }
+        }
+    }
+
+    /**
+     * Sets the sound for the notification.
+     * If a sound is received from the payload, the same is used.
+     * If a sound is not received from the payload, the default sound is used
```

Comment:
can we add info on the format of the sound uri used? e.g. is it just `filename` or `filename.ext`.